src/infra/ingest-datafusion/src/writer.rs

```diff
@@ -89,6 +91,10 @@ impl DataWriterDataFusion {
         self.meta.source_event.as_ref()
     }
 
+    pub fn set_session_context(&mut self, ctx: SessionContext) {
+        self.ctx = ctx;
+    }
```

Comment:
Not ideal, but this is the best way I see to fix this for now.
I have some ideas for refactoring the writer type soon, so we'll see if this extra complexity goes away with it.